File: ruby/lib/Sikaru/run_webhooks/types/webhook_input.rb
# frozen_string_literal: true

module Sikaru
  module RunWebhooks
    module Types
      class WebhookInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :idempotency_key, -> { String }, optional: true, nullable: false

        field :run_id, -> { String }, optional: false, nullable: false

        field :url, -> { String }, optional: false, nullable: false
      end
    end
  end
end

File: typescript/src/api/resources/runWebhooks/client/requests/WebhookInput.ts

/**
 * @example
 *     {
 *         run_id: "run_id",
 *         url: "url"
 *     }
 */
export interface WebhookInput {
    idempotency_key?: string | null;
    run_id: string;
    url: string;
}
